clientd/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+use fedimint_client::{Client, UserClientConfig};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use serde_json::json;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use strum::EnumIter;
+use strum::IntoEnumIterator;
+use tokio::sync::{broadcast, mpsc, oneshot};
+
+mod rpc;
+pub mod server;
+
+// TODO: remove pub and use this type to enforce invariants
+#[derive(Debug, Serialize)]
+pub struct CallbackResponse(pub Result<String, RpcError>);
+
+// TODO: figure out best approach for error handling
+#[derive(Debug, Serialize)]
+pub enum RpcError {
+    ClientError,
+}
+#[derive(Debug)]
+pub enum ManagerMessage {
+    CallHandler(RpcRequest, oneshot::Sender<CallbackResponse>),
+    HandleEvent(Event),
+}
+
+#[derive(Debug)]
+pub enum RpcRequest {
+    HealthCheck(),
+    Info(),
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct Event {
+    event_key: EventKey,
+    data: String,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, EnumIter, Serialize)]
+pub enum EventKey {
+    Mock,
+}
+pub async fn run_manager(
```

Comment:
Manager is typically a bad name because it's so generic. In this particular instance it feels more like a message dispatcher task.

Reply:
mhm..so `run_message_dispatcher()` ?

Reply:
for example